host/xtest/asym_perf.c

```diff
@@ -0,0 +1,944 @@
+			   OAEP_OTHER_LEN)) {
+			fprintf(stderr, "The size or algo is not valid!\n");
+			return -1;
+		}
+	} else {
+		return -1;
+	}
+
+	return 0;
+}
+
+static int check_rsa_hash_params(uint32_t rsa_algo, int width_bits, int size,
+				 int salt_len)
+{
+	int width_bytes = BITS_TO_BYTES(width_bits);
+	int salt_temp = 0;
+	int hash_len = get_rsa_hash_len(rsa_algo);
+
+	if (hash_len == -1)
+		return -1;
+
+	switch (rsa_algo) {
+	case RSASSA_PKCS1_V1_5_SHA1:
+		if (width_bytes < hash_len + DERCODE_SHA1_LEN + PKCS_V1_5_MIN) {
+			fprintf(stderr, "The size or algo is not valid!\n");
+			return -1;
+		}
+		return 0;
+	case RSASSA_PKCS1_V1_5_SHA224:
+	case RSASSA_PKCS1_V1_5_SHA256:
+	case RSASSA_PKCS1_V1_5_SHA384:
+	case RSASSA_PKCS1_V1_5_SHA512:
+		if (width_bytes < hash_len + DERCODE_SHA_LEN + PKCS_V1_5_MIN) {
+			fprintf(stderr, "The size or algo is not valid!\n");
+			return -1;
+		}
+		return 0;
+	case RSASSA_PKCS1_PSS_MGF1_SHA1:
+	case RSASSA_PKCS1_PSS_MGF1_SHA224:
+	case RSASSA_PKCS1_PSS_MGF1_SHA256:
+	case RSASSA_PKCS1_PSS_MGF1_SHA384:
+	case RSASSA_PKCS1_PSS_MGF1_SHA512:
+		salt_temp = (salt_len == 0 ? hash_len : salt_len);
+		if (salt_temp > width_bytes ||
+		    width_bytes < hash_len + salt_temp + PSS_OTHER_LEN) {
+			fprintf(stderr, "The size or algo is not valid!\n");
+			return -1;
+		}
+		return 0;
+	default:
+		return -1;
+	}
+}
+
+int asym_perf_runner_cmd_parser(int argc, char *argv[])
+{
+	int verbosity = CRYPTO_DEF_VERBOSITY;
+	int is_random = CRYPTO_USE_ZEROS;
+	unsigned int n = CRYPTO_DEF_COUNT;
+	unsigned int l = CRYPTO_DEF_LOOPS;
+	int warmup = CRYPTO_DEF_WARMUP;
+	int mode = MODE_GENKEYPAIR;
+	int width_bits = 2048;
+	uint32_t main_algo = RSA;
+	int rsa_algo = -1;
+	int size = 256;
+	int salt_len = 128;
+	int i = 0;
+
+	/* Parse command line */
+	for (i = 1; i < argc; i++) {
+		if (!strcmp(argv[i], "-h") || !strcmp(argv[i], "--help")) {
+			USAGE();
+			return 0;
+		}
+	}
+
+	for (i = 1; i < argc; i++) {
+		if (!strcmp(argv[i], "-a")) {
+			NEXT_ARG(i);
+			if (!strcasecmp(argv[i], "DH"))
+				main_algo = DH;
+			else if (!strcasecmp(argv[i], "RSA_GENKEYPAIR"))
+				main_algo = RSA;
+			else if (!strcasecmp(argv[i], "RSA_NOPAD_ENCRYPT")) {
+				main_algo = RSA;
+				mode = MODE_ENCRYPT;
+				rsa_algo = RSA_NOPAD;
+			} else if (!strcasecmp(argv[i], "RSA_NOPAD_DECRYPT")) {
+				main_algo = RSA;
+				mode = MODE_DECRYPT;
+				rsa_algo = RSA_NOPAD;
+			} else if (!strcasecmp(argv[i], "RSAES_PKCS1_V1_5_ENCRYPT")) {
+				main_algo = RSA;
+				mode = MODE_ENCRYPT;
+				rsa_algo = RSAES_PKCS1_V1_5;
+				size -= PKCS_V1_5_MIN;
+			} else if (!strcasecmp(argv[i], "RSAES_PKCS1_V1_5_DECRYPT")) {
+				main_algo = RSA;
+				mode = MODE_DECRYPT;
+				rsa_algo = RSAES_PKCS1_V1_5;
+				size -= PKCS_V1_5_MIN;
+			} else if (!strcasecmp(argv[i], "RSAES_PKCS1_OAEP_SHA1_ENCRYPT")) {
+				main_algo = RSA;
+				mode = MODE_ENCRYPT;
+				rsa_algo = RSAES_PKCS1_OAEP_SHA1;
+				size -= (OAEP_HASH_LEN(SHA1_LEN) + OAEP_OTHER_LEN);
```

Comment:
could remove the outer parentheses.

ditto at lines 750, 755, 760, 765, 770, .. 790.